integration/network/network.go

```diff
@@ -0,0 +1,173 @@
+package network
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/giantswarm/ipam"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	EnvCircleBuildNumber = "CIRCLE_BUILD_NUM"
+	EnvE2ECIDR           = "E2E_CIDR"
+	EnvE2EMask           = "E2E_MASK"
+
+	EnvAzureCIDR             = "AZURE_CIDR"
+	EnvAzureCalicoSubnetCIDR = "AZURE_CALICO_SUBNET_CIDR"
+	EnvAzureMasterSubnetCIDR = "AZURE_MASTER_SUBNET_CIDR"
+	EnvAzureWorkerSubnetCIDR = "AZURE_WORKER_SUBNET_CIDR"
+
+	AzureCalicoSubnetMask = 17
+	AzureMasterSubnetMask = 24
+	AzureWorkerSubnetMask = 24
+)
+
+func init() {
+	var err error
+
+	var azureNetwork *net.IPNet
+	{
+		azureNetwork, err = AzureNetwork()
+		if err != nil {
+			panic(err)
+		}
+		os.Setenv(EnvAzureCIDR, azureNetwork.String())
```

Comment:
Why are we setting so many env vars? IMO we should try to use the process environment and less as possible. I see that it is necessary for initialization and such but once we run our code we should stay there. This is also why we want to migrate from shell out magic to use appr and helm client libs. We quickly run into situations in which we want to deal with concurrency or some such and then we are trapped. So I would like to approach this here via getters which are only set up via the env. See also https://github.com/giantswarm/aws-operator/blob/c13a7eeb4a574fc5678dfed969c9cc373107f10c/integration/env/env.go#L102. When we have getters for the test relevant configuration we can inject it where needed and we do not need to assume there is data in the process environment.

Reply:
Good catch, I wanted in fact to move this into `AzureNetwork` and `AzureSubnet` logic, so they would only setup the env vars if they are empty.

I do not really understand how your approach suggestion with getters. Could you elaborate on this.

Reply:
I think we can use the information we computed here and dispatch where necessary without writing them into the process environment. Lets drop the `os.Setenv` magic and have a getter instead that returns the value of a variable we initialize here. I provided examples in the link of the aws-operator. Please ask more specific questions when things are unclear I feel like I cannot be more specific from the top of my head now.

Reply:
I did not manage to come up with a more elegant way of passing those CIDR values around. since they are read from environment by e2e-harness.